packages/client-sdk-nodejs/src/internal/storage-data-client.ts

```diff
@@ -0,0 +1,337 @@
+      `Creating leaderboard client using endpoint: '${this.credentialProvider.getStorageEndpoint()}'`
+    );
+
+    // NOTE: This is hard-coded for now but we may want to expose it via StorageConfiguration in the
+    // future, as we do with some of the other clients.
+    const grpcConfig = new StaticGrpcConfiguration({
+      deadlineMillis: this.configuration
+        .getTransportStrategy()
+        .getGrpcConfig()
+        .getDeadlineMillis(),
+      maxSessionMemoryMb: StorageDataClient.DEFAULT_MAX_SESSION_MEMORY_MB,
+    });
+    const channelOptions = grpcChannelOptionsFromGrpcConfig(grpcConfig);
+
+    this.clientWrapper = new IdleGrpcClientWrapper({
+      clientFactoryFn: () =>
+        new store.StoreClient(
+          this.credentialProvider.getStorageEndpoint(),
+          this.credentialProvider.isStorageEndpointSecure()
+            ? ChannelCredentials.createSsl()
+            : ChannelCredentials.createInsecure(),
+          channelOptions
+        ),
+      loggerFactory: this.configuration.getLoggerFactory(),
+      maxIdleMillis: this.configuration
+        .getTransportStrategy()
+        .getMaxIdleMillis(),
+    });
+
+    const context: MiddlewareRequestHandlerContext = {};
+    context[CONNECTION_ID_KEY] = dataClientID;
+    this.interceptors = this.initializeInterceptors(
+      this.configuration.getLoggerFactory(),
+      this.configuration.getMiddlewares(),
+      context
+    );
+  }
+
+  close() {
+    this.logger.debug('Closing storage data clients');
+    this.clientWrapper.getClient().close();
+  }
+
+  private validateRequestTimeout(timeout?: number) {
+    this.logger.debug(`Request timeout ms: ${String(timeout)}`);
+    if (timeout !== undefined && timeout <= 0) {
+      throw new InvalidArgumentError(
+        'request timeout must be greater than zero.'
+      );
+    }
+  }
+
+  private initializeInterceptors(
+    _loggerFactory: MomentoLoggerFactory,
+    middlewares: Middleware[],
+    middlewareRequestContext: MiddlewareRequestHandlerContext
+  ): Interceptor[] {
+    const headers = [
+      new Header('Authorization', this.credentialProvider.getAuthToken()),
+      new Header('Agent', `nodejs:store:${version}`),
+    ];
+    return [
+      middlewaresInterceptor(
+        _loggerFactory,
+        middlewares,
+        middlewareRequestContext
+      ),
+      new HeaderInterceptorProvider(headers).createHeadersInterceptor(),
+      ClientTimeoutInterceptor(this.requestTimeoutMs),
+    ];
+  }
+
+  private createMetadata(storeName: string): Metadata {
+    const metadata = new Metadata();
+    metadata.set('store', storeName);
+    return metadata;
+  }
+
+  public async get(
+    storeName: string,
+    key: string
+  ): Promise<StorageGet.Response> {
+    try {
+      validateStoreName(storeName);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new StorageGet.Error(err)
+      );
+    }
+    this.logger.trace(
```

Comment:
I think we eventually want to move away from `logger.trace` statements towards the per-request logging middleware?
Just calling this out for future work if this is the case